eslint/next.js

```diff
@@ -20,6 +20,18 @@ module.exports = {
     {
       files: JAVASCRIPT_FILES,
       parserOptions: { babelOptions },
+      rules: {
+        // Ignores rule violation for anchors that are children of <Link>s.
+        // See https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/402#issuecomment-368305051
+        'jsx-a11y/anchor-is-valid': [
+          'error',
+          {
+            components: ['Link'],
+            specialLink: ['hrefLeft', 'hrefRight'],
+            aspects: ['invalidHref', 'preferButton'],
+          },
+        ],
+      },
```

Comment:
A side note, the is nested within this config's JavaScript overrides, meaning it would only apply to JavaScript files - not TypeScript.